components/treelist/state.md

```diff
@@ -4,9 +4,875 @@ page_title: TreeList - State
+            {
+               new SortDescriptor()
+               {
+                   Member = nameof(Employee.Id),
+                   SortDirection = ListSortDirection.Descending
+               }
+            },
+            Page = 2
+        };
+
+        args.TreeListState = initialState;
+    }
+
+    public List<Employee> Data { get; set; }
+
+    // sample model
+
+    public class Employee
+    {
+        // hierarchical data collections
+        public List<Employee> DirectReports { get; set; }
+
+        // data fields for display
+        public int Id { get; set; }
+        public string Name { get; set; }
+        public string EmailAddress { get; set; }
+        public DateTime HireDate { get; set; }
+    }
+
+    // data generation
+
+    // used in this example for data generation and retrieval for CUD operations on the current view-model data
+    public int LastId { get; set; } = 1;
+
+    protected override async Task OnInitializedAsync()
+    {
+        Data = await GetTreeListData();
+    }
+
+    async Task<List<Employee>> GetTreeListData()
+    {
+        List<Employee> data = new List<Employee>();
+
+        for (int i = 1; i < 15; i++)
+        {
+            Employee root = new Employee
+            {
+                Id = LastId,
+                Name = $"root: {i}",
+                EmailAddress = $"{i}@example.com",
+                HireDate = DateTime.Now.AddYears(-i),
+                DirectReports = new List<Employee>(), // prepare a collection for the child items, will be populated later in the code
+            };
+            data.Add(root);
+            LastId++;
+
+            for (int j = 1; j < 4; j++)
+            {
+                int currId = LastId;
+                Employee firstLevelChild = new Employee
+                {
+                    Id = currId,
+                    Name = $"first level child {j} of {i}",
+                    EmailAddress = $"{currId}@example.com",
+                    HireDate = DateTime.Now.AddDays(-currId),
+                    DirectReports = new List<Employee>(), // collection for child nodes
+                };
+                root.DirectReports.Add(firstLevelChild); // populate the parent's collection
+                LastId++;
+
+                for (int k = 1; k < 3; k++)
+                {
+                    int nestedId = LastId;
+                    // populate the parent's collection
+                    firstLevelChild.DirectReports.Add(new Employee
+                    {
+                        Id = LastId,
+                        Name = $"second level child {k} of {j} and {i}",
+                        EmailAddress = $"{nestedId}@example.com",
+                        HireDate = DateTime.Now.AddMinutes(-nestedId)
+                    }); ;
+                    LastId++;
+                }
+            }
+        }
+
+        return await Task.FromResult(data);
+    }
+}
+````
+
+### Get The User Action That Changes The TreeList
```

Comment:
Log a task for an example of changing the user action similar to this https://docs.telerik.com/blazor-ui/components/grid/state#get-and-override-user-action-that-changes-the-grid